snapcraft/cli/_options.py

```diff
@@ -52,13 +46,18 @@ def _add_build_options(func):
         for name, params in zip(reversed(_BUILD_OPTION_NAMES),
                                 reversed(_BUILD_OPTIONS)):
             option = click.option(name, **params,
-                                  cls=OptionWithHidden, hidden=hidden)
+                                  cls=HiddenOption if hidden else click.Option)
```

Comment:
The reason it was implemented as it was before as it would be a transparent change once the feature lands in click. That said, this new implementation is more readable and we would need to change code anyways once migrating.